lib/csetperm.gi

```diff
@@ -548,3 +548,148 @@ function(a,b)
   return CanonicalRepresentativeOfExternalSet(a)
          <CanonicalRepresentativeOfExternalSet(b);
 end);
+
+
+
+InstallMethod(Intersection2, "perm groups", IsIdenticalObj,
+  [IsRightCoset and IsPermCollection,IsRightCoset and IsPermCollection],0,
+function(cos1,cos2)
+    local H1, H2, x1, x2, shift, sigma, listMoved_H1, listMoved_H2, listMoved_H12, listMoved_sigma, theInt, set2, set1, eRepr, set2_img, set1_img, H1_sigma, H2_sigma, test, H12, swap, eCos, rho, cosTest, diff12, diff21, fset1, fset2;
+    # We set int = cos1 cap cos2 = H1 x1 cap H2 x2
+    H1:=ActingDomain(cos1);
+    H2:=ActingDomain(cos2);
+    x1:=Representative(cos1);
+    x2:=Representative(cos2);
+    shift:=x1;
+    sigma:=x2 / x1;
+    # Initial very easy check
+    listMoved_H1:=MovedPoints(H1);
+    listMoved_H2:=MovedPoints(H2);
+    listMoved_H12:=Union(listMoved_H1, listMoved_H2);
+    listMoved_sigma:=MovedPoints(sigma);
+    if not IsSubset(listMoved_H12, listMoved_sigma) then
```

Comment:
Hm. I guess there are more necessary conditions; e.g. I think also the union of the points moved by H2 and by sigma must contain the points moved by H1. And the same but with H1 and H2 swapped. Not sure whether any of these imply the other?

But then I think you can do better by analyzing orbits: If `x` is an element of the intersection, then for all positive integers `n`, we have `n^x` is an element of  `n^H1` but also of `(n^H2)^\sigma`. I think this condition strengthens all of the above. We would have:
```suggestion
    listMoved_all := Union(listMoved_H12, listMoved_sigma);
    if ForAny(listMoved_all, n -> IsEmpty(Intersection(Orbit(H1,n), OnSets(Orbit(H2,n),sigma)))) then
```
Of course that's a lazy implementation in that it wastes efforts by recomputing all orbits. But I'll leave optimizations to later.

---

For all this, I wonder: aren't we essentially replicating work that partition backtrack also does, just in a more adhoc fashion?

Reply:
It could be related to the Partition Backtrack, but if so I am unsure how this can be done.
It is certainly true that the intersection of subgroups is using partition backtrack. I do not
know how this could be implemented for the intersection of cosets.
More specific help would be appreciated.

Reply:
Ok, so I added your test. I think now that everything is covered.